From f5bd85347c7ec87bd843de1862679be7a582b940 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:35 -0400
Subject: [ethernet] i40e: Define a new state variable to keep track of feature auto disable

Message-id: <1409658042-22689-35-git-send-email-sassmann@redhat.com>
Patchwork-id: 90228
O-Subject: [RHEL7.1 PATCH 034/281] i40e: Define a new state variable to keep track of feature auto disable
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

This variable is a bit mask. It is needed to differentiate between
user enforced feature disables and auto disable of features due to
HW resource limitations.

Change-ID: Ib4b4f6ae1bb2668c12e482d2555100bc8ad713d5
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 61dade7e9201162cba683cff103cebbdf06655d4)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e.h b/drivers/net/ethernet/intel/i40e/i40e.h
index 838b69b..a191653 100644
--- a/drivers/net/ethernet/intel/i40e/i40e.h
+++ b/drivers/net/ethernet/intel/i40e/i40e.h
@@ -263,6 +263,9 @@ struct i40e_pf {
 #define I40E_FLAG_VXLAN_FILTER_SYNC            (u64)(1 << 27)
 #endif
 
+ /* tracks features that get auto disabled by errors */
+ u64 auto_disable_flags;
+
  bool stat_offsets_loaded;
  struct i40e_hw_port_stats stats;
  struct i40e_hw_port_stats stats_offsets;
-- 
1.7.1